### PR TITLE
Remove metrics publishers

### PR DESCRIPTION
**Pull request by ReubenBond**
This partially addresses #3976

Snapshots of current system behavior are published using various interfaces. This removes that publishing mechanism.

### CODE REVIEW

**Comment by sergeybykov** [comment]
@dotnet-bot test functional

File: src/Orleans.Core/Statistics/ClientStatisticsManager.cs [hunk 1]
```diff
@@ -3,47 +3,36 @@
...
-        private readonly ILoggerFactory loggerFactory;
-        private MonitoringStorageOptions storageOptions;
-        private string dnsHostName;
+        private readonly MonitoringStorageOptions storageOptions;
+        private readonly string dnsHostName;
         public ClientStatisticsManager(
             IOptions<MonitoringStorageOptions> storageOptions,
             SerializationManager serializationManager, 
             IServiceProvider serviceProvider,
-            IHostEnvironmentStatistics hostEnvironmentStatistics,
```
Comment (xiazen) [inline src/Orleans.Core/Statistics/ClientStatisticsManager.cs]: why removing this two. Doesn't statistics still reports cpu, memory and all that ?

Comment (xiazen) [reply src/Orleans.Core/Statistics/ClientStatisticsManager.cs]: Oh I guess these two are here only for constructing the default metric publisher.
Can you make sure statistics still reports `IHostEnvironmentStatistics ` and `IAppEnvironmentStatistics ` ? since those are pretty important statistics

File: src/Orleans.Core/Statistics/ClientTableStatistics.cs [hunk 2]
```diff
@@ -1,142 +0,0 @@
-#define LOG_MEMORY_PERF_COUNTERS 
-using System;
-using System.Threading.Tasks;
-using Microsoft.Extensions.Logging;
-using Orleans.Statistics;
-
-namespace Orleans.Runtime
-{
-    internal class ClientTableStatistics : IClientPerformanceMetrics, IDisposable
```
Comment (xiazen) [inline src/Orleans.Core/Statistics/ClientTableStatistics.cs]: why is this removed? isn't this statistics, not metrics?

File: src/Orleans.Core/Statistics/ClientStatisticsManager.cs (hunk 1 above)
Comment (ReubenBond) [reply src/Orleans.Core/Statistics/ClientStatisticsManager.cs]: These values are only published to the metrics table, so we can remove them since they're no longer used.

File: src/Orleans.Core/Statistics/ClientTableStatistics.cs (hunk 2 above)
Comment (ReubenBond) [reply src/Orleans.Core/Statistics/ClientTableStatistics.cs]: The class name is statistics, but the interface is metrics. This is only used for snapshot metrics

**Comment by ReubenBond** [comment]
Fixed test: metrics & stats values were mixed up in the test grain.

**Comment by ReubenBond** [comment]
Simplified some classes and refactored a little, including removing something which has bothered me for a long time:
`Silo.cs:`
```diff
-            siloStatistics.MetricsTable.ActivationDirectory = activationDirectory;
-            siloStatistics.MetricsTable.ActivationCollector = catalog.ActivationCollector;
-            siloStatistics.MetricsTable.MessageCenter = messageCenter;
```

File: src/Orleans.Core/Statistics/IPerformanceMetrics.cs [hunk 3]
```diff
@@ -29,60 +28,13 @@ public interface ICorePerformanceMetrics
...
-        /// are used for different grain types).
-        /// </summary>
-        int RecentlyUsedActivationCount { get; }
-
-        /// <summary>
-        /// Number of currently connected clients
-        /// </summary>
-        long ClientCount { get; }
-
+    {
```
Comment (xiazen) [inline src/Orleans.Core/Statistics/IPerformanceMetrics.cs]: why do we remove IClientPerformanceMetrics and keep `ISiloPerformanceMetrics ` and `ICorePerformanceMetrics ` ? 

Especially `ICorePerformanceMetrics`, all of whose metrics are covered by `SiloRuntimeStatistics` ? 

Is it because other component is using it , such as Scheduler and WorkerPool? If yes, can they use `IAppEnvironmentStatistics` and `IHostEnvironmentStatistics` instead ?

Comment (ReubenBond) [reply src/Orleans.Core/Statistics/IPerformanceMetrics.cs]: My bad, I explained it here: https://github.com/dotnet/orleans/issues/3976#issuecomment-364274529

**Comment by ReubenBond** [comment]
@dotnet-bot test functional

**Comment by ReubenBond** [comment]
@xiazen I removed the class entirely now

**Comment by sergeybykov** [comment]
@dotnet-bot test functional
